x-pack/plugins/upgrade_assistant/__jest__/client_integration/overview/upgrade_step/upgrade_step.test.tsx

```diff
@@ -33,28 +39,64 @@ describe('Overview - Upgrade Step', () => {
   });
 
   describe('On Cloud', () => {
-    test('Shows upgrade CTA and link to docs', async () => {
-      await act(async () => {
-        testBed = await setupOverviewPage({
-          plugins: {
-            cloud: {
-              isCloudEnabled: true,
-              deploymentUrl:
-                'https://cloud.elastic.co./deployments/bfdad4ef99a24212a06d387593686d63',
-            },
-          },
-        });
+    test('When ready for upgrade, shows upgrade CTA and link to docs', async () => {
+      httpRequestsMockHelpers.setGetUpgradeStatusResponse({
+        readyForUpgrade: true,
+        details: 'Ready for upgrade',
       });
 
-      const { component, exists, find } = testBed;
+      testBed = await setupCloudOverviewPage();
+      const { exists, find, component } = testBed;
       component.update();
 
       expect(exists('upgradeSetupDocsLink')).toBe(true);
       expect(exists('upgradeSetupCloudLink')).toBe(true);
 
+      expect(find('upgradeSetupCloudLink').props().disabled).toBe(false);
       expect(find('upgradeSetupCloudLink').props().href).toBe(
-        'https://cloud.elastic.co./deployments/bfdad4ef99a24212a06d387593686d63?show_upgrade=true'
+        `${DEPLOYMENT_URL}?show_upgrade=true`
+      );
+    });
+
+    test('When not ready for upgrade, the CTA button is disabled', async () => {
+      httpRequestsMockHelpers.setGetUpgradeStatusResponse({
+        readyForUpgrade: false,
+        details: 'Resolve critical deprecations firs',
+      });
+
+      testBed = await setupCloudOverviewPage();
+      const { exists, find, component } = testBed;
+      component.update();
+
+      expect(exists('upgradeSetupDocsLink')).toBe(true);
+      expect(exists('upgradeSetupCloudLink')).toBe(true);
+
+      expect(find('upgradeSetupCloudLink').props().disabled).toBe(true);
+    });
+
+    test('An error callout is displayed, if status check failed', async () => {
+      httpRequestsMockHelpers.setGetUpgradeStatusResponse(
+        {},
+        { statusCode: 500, message: 'Status check failed' }
       );
+
+      testBed = await setupCloudOverviewPage();
+      const { exists, component } = testBed;
+      component.update();
+
+      expect(exists('upgradeSetupDocsLink')).toBe(false);
+      expect(exists('upgradeSetupCloudLink')).toBe(false);
+      expect(exists('upgradeStatusErrorCallout')).toBe(true);
+    });
+
+    test('The CTA button displays loading indicator', async () => {
+      setServerAsync(true);
+      testBed = await setupCloudOverviewPage();
+      const { exists, find } = testBed;
+
+      expect(exists('upgradeSetupDocsLink')).toBe(true);
+      expect(exists('upgradeSetupCloudLink')).toBe(true);
+      expect(find('upgradeSetupCloudLink').childAt(0).props().isLoading).toBe(true);
     });
   });
```

Comment:
nit: I think we could also add a small test to make sure that the retry button also triggers a resend request